tests/e2e/tailscale/Vagrantfile

```diff
@@ -45,13 +45,23 @@ def provision(vm, roles, role_num, node_num)
     end
   end
   if roles.include?("agent")
+    vpn_auth = nil
+    vpn_auth_method = nil
+    if role_num == 0
+      vpn_auth_method = "vpn-auth"
+      vpn_auth = "name=tailscale,joinKey=#{TAILSCALE_KEY}"
+    else
+      vpn_auth_method = "vpn-auth-file"
+      File.write("./vpn-auth-file", vpn_auth)
+      vpn_auth = File.expand_path("./vpn-auth-file")
+    end
     vm.provision :k3s, run: 'once' do |k3s|
       k3s.config_mode = '0644' # side-step https://github.com/k3s-io/k3s/issues/4321
       k3s.args = "agent "
       k3s.config = <<~YAML
         server: https://TAILSCALEIP:6443
         token: vagrant
-        vpn-auth: "name=tailscale,joinKey=#{TAILSCALE_KEY}"
+        #{vpn_auth_method}: #{vpn_auth}
```

Comment:
The file is created in the local node and never scp into the VM, therefore, agent-1 will never be able to join the cluster

Reply:
could you provide an example of how I could do this? I am not familiar with the syntax but the code is in `ruby` right?